File: SieveOfErathosthenes/sieve.c
#include <stdbool.h>
#include <stdio.h>
#include <stdlib.h>
#include <string.h>



void SieveOfErathosthenes(int n){
    bool primes[11]; //n+1 funktioniert nicht obwohl es seit C99 funktionieren sollte

    memset(primes, true, sizeof(primes));
    int p = 2;
    while(p*p <= n) {
        
        if (primes[p] == true) {
            int i = p * p;
            while (i <= n){
                primes[i] = false;
                i += p;
            }
        p++;
        }
    }
 
    for (int p = 2; p <= n; p++)
        if (primes[p] == true)
            printf("%d ",p);

}

int main(int argc, char *argv[]) {
    SieveOfErathosthenes(10);

    return 0;
}

File: SieveOfErathosthenes/sieve.java
public class sieve {

    public static void SieveOfErathosthenes(int n){
        boolean primes[] = new boolean[n + 1];
        for (int i = 0; i <= n; i++)
            primes[i] = true;

        int p = 2;
        while(p*p<=n){
            if (primes[p] == true){
                int i = p*p;
                while (i<=n){
                    primes[i] = false;
                    i += p;
                }
            }
            p++;
        }

        for (int i=2; i<=n; i++){
            if(primes[i] == true){
                System.err.println(i);
            }
        }

    }
    public static void main(String[] args){                                                                                                                                                                                                                                                                                     
        SieveOfErathosthenes(1000000);
    }
}
